sosw/components/test/integration/test_secrets_manager_i.py

```diff
@@ -0,0 +1,108 @@
+import boto3
+import json
+import logging
+import time
+import unittest
+import uuid
+import os
+
+logging.getLogger('botocore').setLevel(logging.WARNING)
+
+os.environ["STAGE"] = "test"
+os.environ["autotest"] = "True"
+
+from sosw.components.config import SecretsManager
+
+TEST_SUFFIX = None
+
+
+def _get_autotest_secret_suffix():
+    """ Get unique suffix for autotest secrets for this run of TestCase """
+
+    global TEST_SUFFIX
+
+    if TEST_SUFFIX is None:
+        TEST_SUFFIX = str(uuid.uuid4())[:18]
+
+    return TEST_SUFFIX
+
+
+class SecretsManagerIntegrationTestCase(unittest.TestCase):
+    TEST_CONFIG = {
+    }
+
+    TEST_SECRETS = [
+        {
+            'Name': f'autotest_first_secret_{_get_autotest_secret_suffix()}',
+            'SecretString': json.dumps({'username': 'test'}),
+            'Tags': [{'Key': 'test_env', 'Value': f'test_first_{_get_autotest_secret_suffix()}'}]
+
+        },
+        {
+            'Name': f'autotest_second_secret_{_get_autotest_secret_suffix()}',
+            'SecretString': json.dumps({'username': 'test'}),
+            'Tags': [{'Key': 'test_env', 'Value': f'test_second_{_get_autotest_secret_suffix()}'}]
+
+        },
+        {
+            'Name': f'autotest_third_secret_{_get_autotest_secret_suffix()}',
+            'SecretString': json.dumps({'username': 'test'}),
+            'Tags': [{'Key': 'test_env', 'Value': f'test_third_{_get_autotest_secret_suffix()}'}]
+
+        }
+    ]
+
+    client = None
+
+    def _get_boto3_client(self) -> boto3.client:
+
+        if not self.client:
+            self.client = boto3.client('secretsmanager')
+
+        return self.client
+
+
+    @classmethod
+    def setUpClass(cls) -> None:
+
+        client = boto3.client('secretsmanager')
+        for test in cls.TEST_SECRETS:
+            client.create_secret(**test)
+
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+
+        client = boto3.client('secretsmanager')
+        for test in cls.TEST_SECRETS:
+            client.delete_secret(SecretId=test['Name'])
+
+
+    def setUp(self):
+
+        self.secretsmanager_obj = SecretsManager()
+
+
+    def tearDown(self):
+        pass
+
+
+    def test_get_secrets_credentials_by_name(self):
+        time.sleep(10)
```

Comment:
Don't sleep here. In case you really have to sleep, sleep in the setUp, although the really correct way is not to sleep, but to check for setUp to complete within the setUp itself.

Reply:
no idea how to do it. for now i add sleep to `setUp`. tomorrow please explain me.